simln-lib/src/sim_node.rs

```diff
@@ -1007,17 +1004,18 @@ impl SimGraph {
 }
 
 /// Produces a map of node public key to lightning node implementation to be used for simulations.
-pub async fn ln_node_from_graph(
+pub async fn ln_node_from_graph<'a>(
+    nodes_info: &HashMap<PublicKey, NodeInfo>,
```

Comment:
Do we need to pass `nodes_info` in here? Could use `graph.lock().await.list_nodes()`